frontend/src/components/elements/PlotlyChart/CustomTheme.tsx

```diff
@@ -0,0 +1,640 @@
+/**
+ * Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { merge, assign } from "lodash"
+
+import { useTheme } from "@emotion/react"
+
+import { hasLightBackgroundColor, Theme } from "src/theme"
+
+// TODO: for these colors below, these likely need to move to our theme!
+// For the meantime, these colors will be defined for plotly.
+
+const divergingColorscaleLightTheme = [
+  [0.1, "#004280"],
+  [0.2, "#0054A3"],
+  [0.3, "#1C83E1"],
+  [0.4, "#60B4FF"],
+  [0.5, "#A6DCFF"],
+  [0.6, "#FFC7C7"],
+  [0.7, "#FF8C8C"],
+  [0.8, "#FF4B4B"],
+  [0.9, "#BD4043"],
+  [1.0, "#7D353B"],
+]
+
+const divergingColorscaleDarkTheme = [
+  [0, "#A6DCFF"],
+  [0.1, "#A6DCFF"],
+  [0.2, "#60B4FF"],
+  [0.3, "#1C83E1"],
+  [0.4, "#0054A3"],
+  [0.5, "#004280"],
+  [0.6, "#7D353B"],
+  [0.7, "#BD4043"],
+  [0.8, "#FF4B4B"],
+  [0.9, "#FF8C8C"],
+  [1.0, "#FFC7C7"],
+]
+
+const sequentialColorscaleLightTheme = [
+  [0, "#E4F5FF"],
+  [0.1111111111111111, "#C7EBFF"],
+  [0.2222222222222222, "#A6DCFF"],
+  [0.3333333333333333, "#83C9FF"],
+  [0.4444444444444444, "#60B4FF"],
+  [0.5555555555555556, "#3D9DF3"],
+  [0.6666666666666666, "#1C83E1"],
+  [0.7777777777777778, "#0068C9"],
+  [0.8888888888888888, "#0054A3"],
+  [1, "#004280"],
+]
+
+const sequentialColorscaleDarkTheme = [
+  [0, "#004280"],
+  [0.1111111111111111, "#0054A3"],
+  [0.2222222222222222, "#0068C9"],
+  [0.3333333333333333, "#1C83E1"],
+  [0.4444444444444444, "#3D9DF3"],
+  [0.5555555555555556, "#60B4FF"],
+  [0.6666666666666666, "#83C9FF"],
+  [0.7777777777777778, "#A6DCFF"],
+  [0.8888888888888888, "#C7EBFF"],
+  [1, "#E4F5FF"],
+]
+
+const categoryColorsLightTheme = [
+  "#0068C9",
+  "#83C9FF",
+  "#FF2B2B",
+  "#FFABAB",
+  "#29B09D",
+  "#7DEFA1",
+  "#FF8700",
+  "#FFD16A",
+  "#6D3FC0",
+  "#D5DAE5",
+]
+
+const categoryColorsDarkTheme = [
+  "#83C9FF",
+  "#0068C9",
+  "#FFABAB",
+  "#FF2B2B",
+  "#7DEFA1",
+  "#29B09D",
+  "#FFD16A",
+  "#FF8700",
+  "#6D3FC0",
+  "#D5DAE5",
+]
+
+export function getGray70(theme: Theme): string {
```

Comment:
nit: I don't think these methods need the `export` since they are only internally used.